docs: add docs (#21)
getoutreach/stencil-golang: docs/arguments.md

# Arguments

Below is a list of all of the arguments that `stencil-golang` supports.

## `service`

**Type**: `bool`
**Default**: `false`

Indicates that this application is a service and that the [service-activity](./service-activity.md) framework should be built.

```yaml
service: true
```

## `serviceActivities`

**Type**: `list`
**Default**: `[]`
**Options**: `['grpc', 'http', 'kafka']`

> A list of service activities that should be generated when `service` is `true`.

Reviewer: ```suggestion
A list of service activities that should be generated. Requires `service` to be set to `true`.
```

```yaml
serviceActivities:
  - grpc
  - http
  - kafka
```

## `commands`

**Type**: `list`
**Default**: `[]`

A list of CLIs to generate for this application.

```yaml
commands:
  - name: my-cli
```

## `grpcClients`

**Type**: `list`
**Default**: `[]`
**Options**: `['node', 'ruby']`

> A list of gRPC clients to generate for this application.

Reviewer: ```suggestion
A list of gRPC clients to generate for this application. Requires `grpc` to be in the `serviceActivities` list.
```
